documentation/Spring-native-vs-Quarkus.md

```diff
@@ -0,0 +1,69 @@
+
+## Quarkus
+
+Quarkus is "_A Kubernetes Native Java stack tailored for OpenJDK HotSpot and **GraalVM**, crafted from the best of breed Java libraries and standards_". It is designed around a **container first philosophy**. i.e it is is optimized for low memory usage and fast startup times. 
+
+Quarkus achieves it in following ways:
+
+* First Class Support for GraalVM :
+
+  Use of AOT compilation
+* Build Time Metadata Processing : 
+
+  As much processing as possible is done at build time, so your application will only contain the classes that are actually needed at runtime. This results in less memory usage, and also faster startup time as all metadata processing has already been done.
+* Reduction in Reflection Usage:
+
+  As much as possible Quarkus tries to avoid reflection, reducing startup time and memory usage.
+* Native Image Pre Boot:
+
+  When running in a native image Quarkus pre-boots as much of the framework as possible during the native image build process. This means that the resulting native image has already run most of the startup code and serialized the result into the executable, resulting in even faster startup.
+
+## Spring- native:
+
+Current version of Spring Native 0.9.1 is designed to be used with Spring Boot 2.4.4
+
+Spring Native provides support for compiling Spring applications to native executables using the GraalVM native-image compiler. Using native image provides key advantages, such as instant startup, instant peak performance, and reduced memory consumption. But there are some disadvantages like it is a heavy process that is slower than a regular application. A native image has fewer runtime optimizations after warmup. Finally, it is less mature than the JVM with some different behaviors.
+
+Difference in regular JVM vs GraalVM:
+
+* A static analysis of your application from the main entry point is performed at build time.
+
+* The unused parts are removed at build time.
+
+* Configuration is required for reflection, resources, and dynamic proxies.
+
+* Classpath is fixed at build time.
+
+* No class lazy loading: everything shipped in the executables will be loaded in memory on startup.
+
+* Some code will run at build time.
+
+* There are some [limitations](https://github.com/oracle/graal/blob/master/substratevm/Limitations.md) around some aspects of Java applications that are not fully supported.
+
+## Build time and start time for apps:
+
+| component| Build time | Start time |
+| ----------- | ----------- | ----------- |
+| Spring native | 19.615s | 2.913s |
+| Quarkus | 52.818s | 0.802s |
+
+## Memory footprints:
+
+TODO
```

Comment:
I think there are even other tests, we should link them here accordinly instead of providing our own.